modules/users/tests/server/user.server.model.tests.js

```diff
@@ -217,9 +216,87 @@ describe('User Model Unit Tests:', function () {
         });
       });
     });
-
   });
 
+  describe("User Password Validation Tests", function() {
+    it('should validate when the password strength passes - "P@$$w0rd!!"', function () {
+      var _user1 = new User(user1);
+      _user1.password = 'P@$$w0rd!!';
+
+      _user1.validate(function (err) {
+        should.not.exist(err);
+      });
+    });
+
+    it('should validate when the passphrase strength passes - "Open-Source Full-Stack Solution For MEAN Applications"', function () {
+      var _user1 = new User(user1);
+      _user1.password = 'Open-Source Full-Stack Solution For MEAN Applications';
+
+      _user1.validate(function (err) {
+        should.not.exist(err);
+      });
+    });
+
+    it('should not allow a less than 10 characters long - "P@$$w0rd!"', function (done) {
+      var _user1 = new User(user1);
+      _user1.password = 'P@$$w0rd!';
+
+      _user1.validate(function (err) {
+        err.errors.password.message.should.equal("The password must be at least 10 characters long.");
+        done();
+      });
+    });
+
+    it('should not allow a greater than 128 characters long.', function (done) {
+      var _user1 = new User(user1);
+      _user1.password = ')!/uLT="lh&:`6X!]|15o!$!TJf,.13l?vG].-j],lFPe/QhwN#{Z<[*1nX@n1^?WW-%_.*D)m$toB+N7z}kcN#B_d(f41h%w@0F!]igtSQ1gl~6sEV&r~}~1ub>If1c+';
+
+      _user1.validate(function (err) {
+        err.errors.password.message.should.equal("The password must be fewer than 128 characters.");
+        done();
+      });
+    });
+
+    it('should not allow more than 3 or more repeating characters ', function (done) {
+      var _user1 = new User(user1);
+      _user1.password = 'L0veSexxxSecre+God';
```

Comment:
haha..yeah, I copied the test passwords from the owasp lib [here](https://github.com/nowsecure/owasp-password-strength-test/blob/master/test.js#L30), I can change that to be less *interesting*..